src/hooks/useQueryInSearchParams.ts

```diff
@@ -17,6 +17,11 @@ import { useQueries } from "./queries";
 import { useUrlBar } from "./useUrlBar";
 import { oracleEthersApiList } from "@/contexts";
 
+function isValidInteger(str: string): boolean {
+  const num = parseInt(str, 10);
+  return !isNaN(num) && num.toString() === str.trim();
+}
+
```

Comment:
can this be replaced with ?`Number.isInteger(Number(str))`

Reply:
maybe so!